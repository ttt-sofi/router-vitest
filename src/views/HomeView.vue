<script setup>
import PostsView from './PostsView.vue'
import ContactView from './ContactView.vue'
import AboutView from './AboutView.vue'
</script>

<template>
<main>
  <PostsView />
  <ContactView/>
  <AboutView/>
  <div>
    <h1>Hello Vite!</h1>
  </div>
</main>

</template>
