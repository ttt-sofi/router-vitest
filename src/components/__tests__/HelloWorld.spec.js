import { describe, it, expect } from "vitest";
import { mount } from "@vue/test-utils";
import HomeView from "@/views/HomeView.vue";

describe("HomeView", () => {
  it("DeberÃ­a renderizar correctamente y coincidir con el snapshot", () => {
    // Monta el componente
    const wrapper = mount(HomeView);

    // Verifica el snapshot del HTML renderizado
    expect(wrapper.html()).toMatchSnapshot();
  });
});