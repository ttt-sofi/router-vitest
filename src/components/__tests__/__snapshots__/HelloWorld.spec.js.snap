// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomeView > Deber√≠a renderizar correctamente y coincidir con el snapshot 1`] = `
"<main>
  <div class="posts">
    <h1>Vista de Posts</h1>
    <ul>
      <li>Post 1</li>
      <li>Post 2</li>
      <li>Post 3</li>
      <li>Post 4</li>
    </ul>
  </div>
  <div class="contact">
    <h1>Vista de Contacto</h1>
    <p>Esta es la vista de contacto</p>
  </div>
  <div class="about">
    <h1>Vista de About</h1>
    <p>Esta es la vista de sobre</p>
  </div>
  <div>
    <h1>Hello Vite!</h1>
  </div>
</main>"
`;
